src/psyclone/psyir/transformations/intrinsics/matmul2code_trans.py

```diff
@@ -478,19 +532,37 @@ def _apply_matrix_matrix(node):
         assign = Assignment.create(result_ref.copy(), rhs)
         # Create ii loop and add the above code as a child
         # Work out the bounds
-        lower_bound, upper_bound, step = _get_array_bound(matrix1, 1)
-        # Must be the same as _get_array_bound(matrix2, 0)
+        if len(m1_fr_order) == 0:
+            # If no full ranges, then matrix was specified
+            # without them i.e.: only has two dimensions.
+            pos_last = 1
+        else:
+            pos_last = m1_fr_order[-1]
+        lower_bound, upper_bound, step = _get_array_bound(matrix1, pos_last)
+        # Must be the same as _get_array_bound(matrix2, pos_first)
```

Comment:
I don't understand this comment as there's no use of this fact in the code? (Probably a good thing should we ever generalise this to allow partial ranges.)

Reply:
Addressed in 10d5abbfb758dbaa45439829f974a99504e27227. The comment was previously there without influencing the functionality so I initially just changed the notation, but now I have removed it.